rpc/client.go

```diff
@@ -191,6 +195,46 @@ func (c *Client) Healthy() <-chan struct{} {
 	return c.healthy.Load().(chan struct{})
 }
 
+func (c *Client) isHealthy() bool {
+	select {
+	case <-c.Healthy():
+		return true
+	default:
+	}
+	return false
+}
+
+// WaitHealthy returns the health of the Client. On the first connection of a
+// newly-created Client, WaitHealthy will block for up to Context.HealthWait if
+// its health has not yet been determined.
+func (c *Client) WaitHealthy() bool {
```

Comment:
This method can be simplified a bit by removing the first phase, and removing the `case <-c.Healthy():` clause from the last `select`. (we never need to call `isHealthy()` except after reading from `c.healthReceived`).